api/src/generalized.rs

```diff
@@ -0,0 +1,362 @@
+        Variable(Variable<'a>),
+    }
+
+    impl<'a> From<NamedNode<'a>> for GeneralizedTerm<'a> {
+        fn from(other: NamedNode<'a>) -> GeneralizedTerm<'a> {
+            GeneralizedTerm::NamedNode(other)
+        }
+    }
+
+    impl<'a> From<BlankNode<'a>> for GeneralizedTerm<'a> {
+        fn from(other: BlankNode<'a>) -> GeneralizedTerm<'a> {
+            GeneralizedTerm::BlankNode(other)
+        }
+    }
+
+    impl<'a> From<Literal<'a>> for GeneralizedTerm<'a> {
+        fn from(other: Literal<'a>) -> GeneralizedTerm<'a> {
+            GeneralizedTerm::Literal(other)
+        }
+    }
+
+    impl<'a> From<Variable<'a>> for GeneralizedTerm<'a> {
+        fn from(other: Variable<'a>) -> GeneralizedTerm<'a> {
+            GeneralizedTerm::Variable(other)
+        }
+    }
+
+    impl<'a> From<NamedOrBlankNode<'a>> for GeneralizedTerm<'a> {
+        fn from(other: NamedOrBlankNode<'a>) -> GeneralizedTerm<'a> {
+            match other {
+                NamedOrBlankNode::NamedNode(inner) => GeneralizedTerm::NamedNode(inner),
+                NamedOrBlankNode::BlankNode(inner) => GeneralizedTerm::BlankNode(inner),
+            }
+        }
+    }
+
+    impl<'a> From<Term<'a>> for GeneralizedTerm<'a> {
+        fn from(other: Term<'a>) -> GeneralizedTerm<'a> {
+            match other {
+                Term::NamedNode(inner) => GeneralizedTerm::NamedNode(inner),
+                Term::BlankNode(inner) => GeneralizedTerm::BlankNode(inner),
+                Term::Literal(inner) => GeneralizedTerm::Literal(inner),
+            }
+        }
+    }
+
+    impl<'a> convert::TryFrom<GeneralizedTerm<'a>> for NamedNode<'a> {
+        type Error = StrictRdfError;
+        fn try_from(other: GeneralizedTerm<'a>) -> Result<NamedNode<'a>, StrictRdfError> {
+            match other {
+                GeneralizedTerm::NamedNode(inner) => Ok(inner),
+                GeneralizedTerm::BlankNode(_) => {
+                    Err("Blankd node can not be used as predicate".into())
+                }
+                GeneralizedTerm::Literal(_) => Err("Literal can not be used as predicate".into()),
+                GeneralizedTerm::Variable(_) => Err("Variable can not be converted to Term".into()),
+            }
+        }
+    }
+
+    impl<'a> convert::TryFrom<GeneralizedTerm<'a>> for NamedOrBlankNode<'a> {
+        type Error = StrictRdfError;
+        fn try_from(other: GeneralizedTerm<'a>) -> Result<NamedOrBlankNode<'a>, StrictRdfError> {
+            match other {
+                GeneralizedTerm::NamedNode(inner) => Ok(NamedOrBlankNode::NamedNode(inner)),
+                GeneralizedTerm::BlankNode(inner) => Ok(NamedOrBlankNode::BlankNode(inner)),
+                GeneralizedTerm::Literal(_) => Err("Literal can not be used a subject".into()),
+                GeneralizedTerm::Variable(_) => Err("Variable can not be converted to Term".into()),
+            }
+        }
+    }
+
+    impl<'a> convert::TryFrom<GeneralizedTerm<'a>> for Term<'a> {
```

Comment:
Why not just `use std::convert::TryFrom`?

Reply:
Sorry, I don't understand what you mean here...

Reply:
Sorry, just add a `use std::convert::TryFrom` at the beginning of the file to avoid having to write `convert::TryFrom` each time it's used.